src/auth/auth.go

```diff
@@ -139,44 +149,20 @@ func IsErrExpiredToken(err error) bool {
 }
 
 // ErrNotAuthorized is returned if the user is not authorized to perform
-// a certain operation. Either
-// 1) the operation is a user operation, in which case 'Repo' and/or 'Required'
-// 		should be set (indicating that the user needs 'Required'-level access to
-// 		'Repo').
-// 2) the operation is an admin-only operation (e.g. DeleteAll), in which case
-//    AdminOp should be set
+// a certain operation.
 type ErrNotAuthorized struct {
 	Subject string // subject trying to perform blocked operation -- always set
 
-	Repo     string // Repo that the user is attempting to access
-	Required Scope  // Caller needs 'Required'-level access to 'Repo'
-
-	// Group 2:
-	// AdminOp indicates an operation that the caller couldn't perform because
-	// they're not an admin
-	AdminOp string
+	Resource Resource     // Resource that the user is attempting to access
+	Required []Permission // Caller needs 'Required'-level access to 'Resource'
```

Comment:
We no longer have two types of permissions errors - instead we return the resource and the set of missing permissions required.